mir_eval/melody.py

```diff
@@ -306,130 +295,95 @@ def to_cent_voicing(ref_time, ref_freq, est_time, est_freq, base_frequency=10.,
-    (prediction).  The sequences must be of the same length.
-
-    Examples
-    --------
-    >>> ref_time, ref_freq = mir_eval.io.load_time_series('ref.txt')
-    >>> est_time, est_freq = mir_eval.io.load_time_series('est.txt')
-    >>> (ref_v, ref_c,
-    ...  est_v, est_c) = mir_eval.melody.to_cent_voicing(ref_time,
-    ...                                                  ref_freq,
-    ...                                                  est_time,
-    ...                                                  est_freq)
-    >>> recall, false_alarm = mir_eval.melody.voicing_measures(ref_v,
-    ...                                                        est_v)
-
-    Parameters
-    ----------
-    ref_voicing : np.ndarray
-        Reference boolean voicing array
-    est_voicing : np.ndarray
-        Estimated boolean voicing array
+def voicing_recall(ref_voicing, est_voicing):
+    validate_voicing(ref_voicing, est_voicing)
+    if ref_voicing.size == 0 or est_voicing.size == 0:
+        return 0.
+    ref_indicator = (ref_voicing > 0).astype(float)
+    if np.sum(ref_indicator) == 0:
+        return 1
+    return np.sum(est_voicing * ref_indicator) / np.sum(ref_indicator)
 
-    Returns
-    -------
-    vx_recall : float
-        Voicing recall rate, the fraction of voiced frames in ref
-        indicated as voiced in est
-    vx_false_alarm : float
-        Voicing false alarm rate, the fraction of unvoiced frames in ref
-        indicated as voiced in est
 
-    """
+def voicing_false_alarm(ref_voicing, est_voicing):
     validate_voicing(ref_voicing, est_voicing)
-    ref_voicing = ref_voicing.astype(bool)
-    est_voicing = est_voicing.astype(bool)
-    # When input arrays are empty, return 0 by special case
     if ref_voicing.size == 0 or est_voicing.size == 0:
         return 0.
+    ref_indicator = (ref_voicing == 0).astype(float)
+    if np.sum(ref_indicator) == 0:
+        return 0
+    return np.sum(est_voicing * ref_indicator) / np.sum(ref_indicator)
 
-    # How voicing is computed
-    #        | ref_v | !ref_v |
-    # -------|-------|--------|
-    # est_v  |  TP   |   FP   |
-    # -------|-------|------- |
-    # !est_v |  FN   |   TN   |
-    # -------------------------
-
-    TP = (ref_voicing*est_voicing).sum()
-    FP = ((ref_voicing == 0)*est_voicing).sum()
-    FN = (ref_voicing*(est_voicing == 0)).sum()
-    TN = ((ref_voicing == 0)*(est_voicing == 0)).sum()
-
-    # Voicing recall = fraction of voiced frames according the reference that
-    # are declared as voiced by the estimate
-    if TP + FN == 0:
-        vx_recall = 0.
-    else:
-        vx_recall = TP/float(TP + FN)
 
-    # Voicing false alarm = fraction of unvoiced frames according to the
-    # reference that are declared as voiced by the estimate
-    if FP + TN == 0:
-        vx_false_alm = 0.
-    else:
-        vx_false_alm = FP/float(FP + TN)
+def hz2cents(freq_hz, base_frequency=10.0):
```

Comment:
It looks like you dropped the docstring for this - any reason why and why it needs to be moved?

Reply:
I'd initially taken it out completely but realized it had to go back in because resampling in cents != resampling in Hz. I'll put it back where it was!